client/src/pages/Map/MapLayout.js

```diff
@@ -53,14 +52,10 @@ function MapLayout() {
     setMapSelectionEnabled(!newTreeState.isDragging);
   }, [newTreeState.isDragging]);
 
-  useEffect(() => {
-    if (currentTreeDb) {
-      setCurrentTreeData({
-        ...currentTreeDb,
-        ...currentTreeDataVector,
-      });
-    }
-  }, [currentTreeDb]);
+  const currentTreeData = {
```

Comment:
Just a small change - this doesn't need to be in state since it only updates when `currentTreeDataVector` or `currentTreeDb` updates